File: 3662_filter_characters_by_frequency/Solution.scala
// LeetCode 3662 - Filter Characters by Frequency
// https://leetcode.com/problems/filter-characters-by-frequency/

object Solution {
  def filterCharacters(s: String, k: Int): String = {
    val cnt = new Array[Int](26)
    for (c <- s) cnt(c - 'a') += 1
    val ans = new StringBuilder
    for (c <- s) if (cnt(c - 'a') < k) ans.append(c)
    ans.toString
  }
}

File: 3662_filter_characters_by_frequency/Solution.kt
// LeetCode 3662 - Filter Characters by Frequency
// https://leetcode.com/problems/filter-characters-by-frequency/

class Solution {
    fun filterCharacters(s: String, k: Int): String {
        var cnt = IntArray(26)
        for (c in s.toCharArray()) { cnt[c - 'a']++ }
        var ans = StringBuilder()
        for (char c : s.toCharArray())
            if (cnt[c - 'a'] < k) ans.append(c)
        return ans.toString()
    }
}
